### ISSUE TITLE
New semantic analyzer: max iteration count reached when base class is bad

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#6892]
This code causes max iteration count error when using the new semantic analyzer:

```py
class C:
    class E: pass

    def f(self) -> None:
        class D(self.E): # Should generate error here
            pass
```
